kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -0,0 +1,357 @@
+                                            "variable_names":[],
+                                            "center":[0,0,0],
+                                            "axis_of_rotation":[0.0,0.0,0.0],
+                                            "angle":0.0,
+                                            "dir_of_translation":[0.0,0.0,0.0],
+                                            "magnitude":0.0
+                                            }  )");
+
+        // Initializing
+        mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+        mCenterOfRotation.push_back(mParameters["center"][0].GetDouble());
+        mCenterOfRotation.push_back(mParameters["center"][1].GetDouble());
+        mCenterOfRotation.push_back(mParameters["center"][2].GetDouble());
+
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][0].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][1].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][2].GetDouble());
+
+        mModulus = mParameters["magnitude"].GetDouble();
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][0].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][1].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][2].GetDouble());
+
+        mTheta = mParameters["angle"].GetDouble() * 2 * 3.1416 / 360.0;
+
+        mTransformationMatrix.resize(4,4);
+        mTransformationMatrixVariable.resize(4,4);
+
+        RemoveCommonNodesFromSlaveModelPart();
+
+        if (mTheta == 0 && mModulus != 0)
+            mType = "translation";
+        else if (mTheta != 0.0 && mModulus == 0.0)
+            mType = "rotation";
+        else
+            KRATOS_ERROR_IF(mTheta == 0.0 && mModulus == 0.0)<<"Both angle of rotation and modulus of translation cannot be zero. Please check the input"<<std::endl;
+        else
+            KRATOS_ERROR_IF(mTheta != 0.0 && mModulus != 0.0)<<"Both angle of rotation and modulus of translation cannot be specified at the same time. Please check the input"<<std::endl;
+
+        CalculateTransformationMatrix();
+        mIsInitialized = false;
+    }
+
+
+    ApplyPeriodicConditionProcess::~ApplyPeriodicConditionProcess()
+    {
+    }
+
+    /**
+     * @brief  Function to remove the common nodes of slave and master modelparts from the slave modelpart
+     */
+    void ApplyPeriodicConditionProcess::RemoveCommonNodesFromSlaveModelPart()
+    {
+        const int num_slave_nodes = mrSlaveModelPart.NumberOfNodes();
+        const ModelPart::NodeIterator it_slave_node_begin = mrSlaveModelPart.NodesBegin();
+        for(int i_node = 0; i_node<num_slave_nodes; ++i_node)
+        {
+            ModelPart::NodeIterator it_slave_node = it_slave_node_begin;
+            std::advance(it_slave_node, i_node);
+            auto iterator = mrMasterModelPart.Nodes().find(it_slave_node->Id());
```

Comment:
why is this needed?

Reply:
because, since we are finding slave node on the master modelpart, then if there are common nodes on slave and master, then this means we can end up constraining the node (in turn dof) with itself. 
to avoid this, I remove any common nodes from the slave modelpart.

Reply:
why constrainting a node with itself is it a problem? we are doing it with vicente for nodal contact... and it atually works correctly (and it should, since you can actually use it for changing dofs)

Reply:
yes @RiccardoRossi  ... my bad .. I should say not only constraining node with itself but also a dof with itself. with @loumalouomega implementation we are constraining different dofs on the same node. This is not at all a problem.